### DISCUSSION TITLE
How to implement function recursion？

### DISCUSSION THREAD

**Discussion by tt0203** [duckdb/duckdb#11336] [Ideas]
CREATE MACRO fx(x) AS if(x<=1,1,fx(x-1) + fx(x-2));
I want to implement defining a method that can call the function itself within a function expression?
Similar to the Fibonacci sequence described above.

**Comment by Tishj** [comment]
Recursive CTE I think?

**Reply by tt0203** [reply]
thanks.I have got it!

create macro fib(x) as table
with recursive fx(n,a,b) as (select 1,1::hugeint,1::hugeint union all select n+1,b,a+b from fx where n<x)
select a from fx;
select * from fib(100);

**Comment by tt0203** [comment]
create macro fib(x) as table
with recursive fx(n,a,b) as (select 1,1::hugeint,1::hugeint union all select n+1,b,a+b from fx where n<x)
select a from fx;
select * from fib(100);
